engine/components-api/src/com/cloud/agent/AgentManager.java

```diff
@@ -141,7 +142,7 @@
 
     public void pullAgentOutMaintenance(long hostId);
 
-    boolean reconnect(long hostId);
+    void reconnect(long hostId) throws CloudRuntimeException, AgentUnavailableException;
```

Comment:
Hmm, I understand the re-throwing of a checked exception as runtime one.
So, from what I understood, you want to tell other people that may use the "reconnect" method that it throws a cloud runtime exception, is that it?